File: Solutions/0007- Reverse Integer/Solution.py
class Solution:
    def reverse(self, x: int) -> int:
        answer = 0
        max = 2 ** 31 - 1
        min = -(2 ** 31)
        z = x
        x = abs(x)

        while x != 0:
            if answer > max // 10 or answer < min // 10:
                return 0
            answer = (answer * 10) + (x % 10)
            x //= 10

        if z < 0:
            return -answer
        return answer

File: Solutions/0007- Reverse Integer/Solution.java
class Solution {
    public boolean isPalindrome(int x) {
        int temp = x;
        int reverse = 0;
        
        while (x > 0) {
            reverse = (reverse * 10) + (x % 10);
            x /= 10;
        }
        
        return temp == reverse;
    }
}